docs/5.extensibility.md

```diff
@@ -0,0 +1,94 @@
+# 5. Extensibility
```

Comment:
Nit: what if for human readers and bots we call this `5. Hooks`? It's easier to link to, makes the scope/contents self documenting...

(Unrelated, but as we now know this repo is sticking around for a while, we might want to ditch the numbers in the file names and add a TOC to the readme/intro, so we have room to grow the docs without invalidating links. Writing this here so I dont forget).